libs/vkd3d/device.c

```diff
@@ -5673,6 +5673,13 @@ static D3D12_HEAP_PROPERTIES * STDMETHODCALLTYPE d3d12_device_GetCustomHeapPrope
             heap_properties->MemoryPoolPreference = D3D12_MEMORY_POOL_L0;
             break;
 
+        case D3D12_HEAP_TYPE_GPU_UPLOAD:
+            heap_properties->CPUPageProperty = d3d12_device_is_uma(device, &coherent) && coherent
+                    ? D3D12_CPU_PAGE_PROPERTY_WRITE_BACK : D3D12_CPU_PAGE_PROPERTY_WRITE_COMBINE;
+            heap_properties->MemoryPoolPreference = d3d12_device_is_uma(device, NULL)
+                    ?  D3D12_MEMORY_POOL_L0 : D3D12_MEMORY_POOL_L1;
```

Comment:
Weird space after `?`

Reply:
I see this is inherited from the D3D12_HEAP_TYPE_DEFAULT line above. Might as well fix that while you're at it.